lib/rules/no-unused-prop-types.js

```diff
@@ -78,15 +80,16 @@ module.exports = {
     }
 
     /**
-     * Check if we are in a class constructor
+     * Check if we are in a lifecycle method
      * @return {boolean} true if we are in a class constructor, false if not
      **/
-    function inComponentWillReceiveProps() {
+    function inLifeCycleMethod() {
       var scope = context.getScope();
       while (scope) {
         if (
           scope.block && scope.block.parent &&
-          scope.block.parent.key && scope.block.parent.key.name === 'componentWillReceiveProps'
+          scope.block.parent.key &&
+            LIFE_CYCLE_METHODS.includes(scope.block.parent.key.name)
```

Comment:
arrays don't have `.includes` until node 6; we need to use `indexOf` or https://npmjs.com/array-includes here.